ts/Core/Legend/LegendSymbol.ts

```diff
@@ -100,22 +100,31 @@ namespace LegendSymbol {
 
         let attr: SVGAttributes = {},
             legendSymbol,
-            markerOptions = options.marker;
+            markerOptions = options.marker,
+            symbolOffset = 0,
+            lineSizer;
 
         // Draw the line
         if (!this.chart.styledMode) {
             attr = {
-                'stroke-width': options.lineWidth || 0
+                'stroke-width': options.lineWidth || 0,
+                'stroke-linecap': (options.linecap || 'round') as any
             };
+
             if (options.dashStyle) {
                 attr.dashstyle = options.dashStyle;
             }
         }
 
+
+        lineSizer = attr['stroke-width'] ?
+            Math.min(attr['stroke-width'] / 2, symbolWidth / 2) :
+            0;
```

Comment:
In styled mode, the [result will be 0](https://jsfiddle.net/highcharts/2gLrap5t/) because the stroke-width attribute does not exist. This is a common pattern that you can see in for example [Chart.ts](https://github.com/highcharts/highcharts/blob/98e732a40b8737d6199615c181adc2a3b9675a91/ts/Core/Chart/Chart.ts#L2145-L2171). The idea is to 
1. Generate the line first without a path definition. Just an empty `.path()`, you can add class , attr and all as usual.
2. Then read `legendItem.line.strokeWidth()` and set the path.